client-app/src/desktop/tabs/other/routing/AdvancedRoutingPanel.tsx

```diff
@@ -0,0 +1,198 @@
+import {HoistModel, hoistCmp, creates, XH} from '@xh/hoist/core';
+import {grid, GridModel} from '@xh/hoist/cmp/grid';
+import {action, observable, makeObservable} from '@xh/hoist/mobx';
+import {panel} from '@xh/hoist/desktop/cmp/panel';
+import {wrapper} from '../../../common';
+import React from 'react';
+import {Icon} from '@xh/hoist/icon';
+import {filler, hbox, hframe, span, vbox, vframe} from '@xh/hoist/cmp/layout';
+import {colAutosizeButton, refreshButton} from '@xh/hoist/desktop/cmp/button';
+import {select, switchInput} from '@xh/hoist/desktop/cmp/input';
+
+export const advancedRoutingPanel = hoistCmp.factory({
+    displayName: 'AdvancedRoutingPanel',
+    model: creates(() => new AdvancedRoutingPanelModel()),
+
+    render({model}) {
+        return wrapper({
+            description: [
+                <p>
+                    This example demonstrates how to use URL route parameters to store and restore
+                    the state of a component. The state of the grid (grouping, sorting, and selected
+                    record) is stored in the URL, and the state is restored when the URL is
+                    revisited.
+                </p>,
+                <p>
+                    Hoist applications are able to navigate to a specific URL and specify whether or
+                    not to push onto the route history. In this example, selecting individual
+                    records in the grid will not save the URL to the route history, but changing the{' '}
+                    <code>groupBy</code> or <code>sortBy</code> fields will. Hoist also provides the
+                    ability to prevent route deactivation, allowing the developer to present the
+                    user with a pop-up before navigating away from the current route.
+                </p>,
+                <p>
+                    The state is encoded in the URL as a <code>base64</code> string, which is then
+                    decoded and parsed to restore the state.
+                </p>,
+                <p>
+                    The current state encoding is: <br />
+                    <br />
+                    <code>groupBy: {model.groupBy || 'None'}</code>
+                    <br />
+                    <code>sortBy: {model.sortBy || 'None'}</code>
+                    <br />
+                    <code>selectedId: {model.gridModel.selectedRecord?.id || 'None'}</code>
```

Comment:
I had expected this little output to be showing the decoded route params more directly, rather than relaying the state from the model itself.  I think that would be a bit more interesting / direct, to show the decoded route as it exists for the current page